src/modules/currency/formatAmount.ts

```diff
@@ -0,0 +1,45 @@
+const formatAmount = (
+  currency_code: string,
+  amount: string,
+  options: {
+    locale?: string;
+    withSymbol?: boolean;
+    currencyDisplay?: "code" | "symbol" | "narrowSymbol" | "name";
+  } = {}
+): string => {
+  if (!Number(amount)) throw new Error("Parameter `amount` is not a number!");
+
+  const { withSymbol = true, currencyDisplay = "symbol" } = options;
+  let { locale } = options;
+
+  // Set the locale for number formatting
+  const numberFormatOptions: Intl.NumberFormatOptions = {
+    style: withSymbol ? "currency" : "decimal",
+    currency: currency_code,
+  };
+
+  // Define a separate options object for currencyDisplay
+  const currencyDisplayOptions: Record<string, string> = {
+    currencyDisplay: currencyDisplay,
+  };
+
+  // Merge the two options objects
+  const mergedOptions = {
+    ...numberFormatOptions,
+    ...currencyDisplayOptions,
+  };
+
+  // If a specific locale is provided, use it; otherwise, use the browser's locale
+  if (!locale) {
+    locale = window.navigator.language;
```

Comment:
In non-browser environments, such as server-side JavaScript or Node.js, or older browser versions or modifying privacy settings, this property may not be defined. I think adding a fallback can work here or we can even write a polyfill for this.

Reply:
I don't think there's a way to determine locale in nodejs env on our own. This will have to be passed by the consumer. If we are not able to compute locale, we should throw an exception to pass a valid locale.